aws/resource_aws_qldb_ledger.go

```diff
@@ -0,0 +1,286 @@
+	}
+
+	// QLDB stuff
+	if err := d.Set("name", qldbLedger.Name); err != nil {
+		return fmt.Errorf("error setting name: %s", err)
+	}
+
+	if err := d.Set("deletion_protection", qldbLedger.DeletionProtection); err != nil {
+		return fmt.Errorf("error setting deletion protection: %s", err)
+	}
+
+	// ARN
+	if err := d.Set("arn", qldbLedger.Arn); err != nil {
+		return fmt.Errorf("error setting ARN: %s", err)
+	}
+
+	// Tags
+	log.Printf("[INFO] Fetching tags for %s", d.Id())
+	tags, err := keyvaluetags.QldbListTags(conn, d.Get("arn").(string))
+	if err != nil {
+		return fmt.Errorf("Error listing tags for QLDB Ledger: %s", err)
+	}
+
+	if err := d.Set("tags", tags.IgnoreAws().Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
+	}
+
+	// Setting the permissions_mode manually because GO AWS SDK currently
+	// does not return the set permissions_mode
+	if err := d.Set("permissions_mode", qldb.PermissionsModeAllowAll); err != nil {
+		return fmt.Errorf("error setting permissions mode: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsQLDBLedgerUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).qldbconn
+
+	// Turn on partial mode
+	d.Partial(true)
+
+	if d.HasChange("deletion_protection") {
+		val := d.Get("deletion_protection").(bool)
+		modifyOpts := &qldb.UpdateLedgerInput{
+			Name:               aws.String(d.Id()),
+			DeletionProtection: aws.Bool(val),
+		}
+		log.Printf(
+			"[INFO] Modifying deletion_protection QLDB attribute for %s: %#v",
+			d.Id(), modifyOpts)
+		if _, err := conn.UpdateLedger(modifyOpts); err != nil {
+
+			return err
+		}
+
+		d.SetPartial("deletion_protection")
+	}
+
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
+		if err := keyvaluetags.QldbUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
+			return fmt.Errorf("error updating tags: %s", err)
+		}
+	}
+
+	d.Partial(false)
+	return resourceAwsQLDBLedgerRead(d, meta)
+}
+
+func resourceAwsQLDBLedgerDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).qldbconn
+	deleteLedgerOpts := &qldb.DeleteLedgerInput{
+		Name: aws.String(d.Id()),
+	}
+	log.Printf("[INFO] Deleting QLDB Ledger: %s", d.Id())
+
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := conn.DeleteLedger(deleteLedgerOpts)
+
+		if isAWSErr(err, qldb.ErrCodeResourceInUseException, "") {
+			return resource.RetryableError(err)
+		}
+
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		_, err = conn.DeleteLedger(deleteLedgerOpts)
+	}
+
+	if isAWSErr(err, qldb.ErrCodeResourceNotFoundException, "") {
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error deleting QLDB Ledger (%s): %s", d.Id(), err)
+	}
+
+	if err := waitForQLDBLedgerDeletion(conn, d.Id()); err != nil {
+		log.Printf("[ERROR] Error waiting for QLDB Ledger (%s) deletion: %s", d.Id(), err)
```

Comment:
We should return an error here instead of logging 👍 

```suggestion
		return fmt.Errorf("error waiting for QLDB Ledger (%s) deletion: %s", d.Id(), err)
```